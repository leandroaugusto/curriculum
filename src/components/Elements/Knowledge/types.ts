export interface KnowledgeProps {
  readonly name: string;
  readonly value: number;
}
