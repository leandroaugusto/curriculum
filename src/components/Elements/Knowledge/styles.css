.knowledge-list {
  @apply grid gap-x-2 grid-cols-10;
}
.knowledge-list li {
  @apply h-3.5;
}
@media print {
  .knowledge-list li {
    @apply h-2.5;
  }
  .knowledge-list li.filled {
    @apply bg-slate-600;
  }
  .knowledge-list li.empty {
    @apply bg-slate-300;
  }
}
