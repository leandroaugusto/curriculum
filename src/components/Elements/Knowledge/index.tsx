import "./styles.css";

import { KnowledgeProps } from "./types";

const KNOWLEDGE_SIZE = 10;

export default function Knowledge({
  name,
  value,
}: KnowledgeProps): React.ReactElement<HTMLDivElement> {
  return (
    <div className="mb-5">
      <h4 className="font-bold my-2">{name}</h4>
      <ul className="knowledge-list">
        {[...Array(KNOWLEDGE_SIZE)].map((_, index) => (
          <li
            key={index}
            data-testid={`knowledge-item-${index}`}
            className={`${
              index < value ? "filled bg-slate-600" : "empty bg-slate-300"
            }`}
          />
        ))}
      </ul>
    </div>
  );
}
