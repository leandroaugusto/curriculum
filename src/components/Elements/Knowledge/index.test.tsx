import { render, screen } from "@testing-library/react";

import Knowledge from ".";

describe("Knowledge", () => {
  it("renders the Knowledge element", () => {
    const { container } = render(<Knowledge name="name" value={10} />);
    expect(container).toMatchSnapshot();
  });

  it("renders the Knowledge element with fill color", () => {
    render(<Knowledge name="name" value={5} />);

    const lastItemFilled = screen.getByTestId("knowledge-item-4");
    const firstItemEmpty = screen.getByTestId("knowledge-item-5");

    expect(lastItemFilled).toHaveClass("filled bg-slate-600");
    expect(firstItemEmpty).toHaveClass("empty bg-slate-300");
  });
});
