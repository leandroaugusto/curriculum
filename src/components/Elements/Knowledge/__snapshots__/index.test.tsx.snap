// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Knowledge renders the Knowledge element 1`] = `
<div>
  <div
    class="mb-5"
  >
    <h4
      class="font-bold my-2"
    >
      name
    </h4>
    <ul
      class="knowledge-list"
    >
      <li
        class="filled bg-slate-600"
        data-testid="knowledge-item-0"
      />
      <li
        class="filled bg-slate-600"
        data-testid="knowledge-item-1"
      />
      <li
        class="filled bg-slate-600"
        data-testid="knowledge-item-2"
      />
      <li
        class="filled bg-slate-600"
        data-testid="knowledge-item-3"
      />
      <li
        class="filled bg-slate-600"
        data-testid="knowledge-item-4"
      />
      <li
        class="filled bg-slate-600"
        data-testid="knowledge-item-5"
      />
      <li
        class="filled bg-slate-600"
        data-testid="knowledge-item-6"
      />
      <li
        class="filled bg-slate-600"
        data-testid="knowledge-item-7"
      />
      <li
        class="filled bg-slate-600"
        data-testid="knowledge-item-8"
      />
      <li
        class="filled bg-slate-600"
        data-testid="knowledge-item-9"
      />
    </ul>
  </div>
</div>
`;
